packages/amplify-graphql-auth-transformer/src/vtl-generator/rds/resolvers/common.ts

```diff
@@ -89,7 +90,7 @@ const convertAuthRoleToVtl = (role: RoleDefinition, fields: string[], hideAllowe
           type: str(role.strategy),
           provider: str('iam'),
           roleArn: role.strategy === 'public' ? ref('ctx.stash.unauthRole') : ref('ctx.stash.authRole'),
-          cognitoIdentityPoolId: ref('ctx.identity.cognitoIdentityPoolId'),
+          cognitoIdentityPoolId: str(identityPoolId),
```

Comment:
1. Is there a way to use the type system to ensure that we always pass a valid identity pool ID to cases that need it?
2. What runtime guarantees apply here? Do we need to add a VTL null check to ensure that the identityPoolId is not null/undef/empty before authorizing, or does the AppSync auth util method do that?

Reply:
1. This is not possible without a major refactor to AuthTransformer as it currently receive `identityPoolId` through constructor arguments and it is currently optional.
2. If identityPoolId is null, AuthUtil will return null which we handle and throw 'unauthorized' error.